File: frontend/src/types/stock.ts
export interface StockQuote {
  symbol: string;
  company_name: string;
  price: number;
  previous_close: number;
  change: number;
  change_percent: number;
  market_cap: number | null;
  fifty_two_week_high: number | null;
  fifty_two_week_low: number | null;
  currency: string;
}

File: backend/app/models/stock.py
from pydantic import BaseModel


class StockQuote(BaseModel):
    symbol: str
    company_name: str
    price: float
    previous_close: float
    change: float
    change_percent: float
    market_cap: int | None
    fifty_two_week_high: float | None
    fifty_two_week_low: float | None
    currency: str